chapter-4/FavotireGameList/FavotireGameList/FavotireGameList.cpp

```diff
@@ -0,0 +1,156 @@
+﻿#include <iostream>
+#include <string>
+#include <algorithm>
+#include <vector>
+
+using namespace std;
+
+/* Псевдокод
+
+	Приветственное слово К
+	Он будет делать
+	{
+	П может просмотреть игры
+	   Если нет игр, то выводится сообщение
+
+	П может добавить игру
+	   Если игра уже есть в списке - сообщает, что такая игра уже есть
+
+	П может удалить заголовок игры
+	   Если игр вообще нет - выводится сообщение
+	   Если выбранной игры нет - выдаёт сообщение
+	   Если неверно введена игра - выдаёт сообщение о том, что игры не существует
+
+	П может удалить все игры в списке
+	   Если игр вообще нет - выводится сообщение
+	} Пока пользователь не захочет выйти из программы
+
+	Переделать else if на switch
+*/
+
+int main()
+{
+	setlocale(0, "Russian");
+
+	vector<string>gameList;
+	vector<string>::iterator deleteGames;
+	vector<string>::iterator watchGames;
+
+	int user_input;
+	string user_game_input;
+
+	enum actions { WATCH = 1, ADD, DELETE, DELETE_ALL, HELP, QUIT };
+
+	cout << "Добро пожаловать в нашу программу по спику ваших любимых игр" << endl;
+
+	cout << "\nЧто бы вы хотели сделать?" << endl;
+	cout << "Выберите " << WATCH << ", чтобы посмотреть список текущих игр" << endl;
+	cout << "Выберите " << ADD << ", чтобы добавить игру" << endl;
+	cout << "Выберите " << DELETE << ", чтобы удалить игру" << endl;
+	cout << "Выберите " << DELETE_ALL << ", чтобы удалить все игры" << endl;
+	cout << "Выберите " << HELP << ", чтобы получить помощь" << endl;
+	cout << "Если хотите выйти из программы - введите " << QUIT << endl;
+
+	while (true)
+	{
+		cout << "\nВаш выбор: ";
+		cin >> user_input;
+
+		switch (user_input)
+		{
+		case WATCH:
+			if (gameList.empty())
+			{
+				cout << "\nИзвините, но в вашей библиотеке нет игр" << endl;
+			}
+			else
+			{
+				cout << "\nНа данный момент у вас " << gameList.size() << " игр и среди них: \n";
+				for (watchGames = gameList.begin(); watchGames != gameList.end(); ++watchGames)
+				{
+					cout << " " << *watchGames << endl;
+				}
+			}
+			break;
```

Comment:
Мы говорили, что такую конструкцию можно делать более читаемой, уменьшив вложенность:

```suggestion
			if (gameList.empty())
			{
				cout << "\nИзвините, но в вашей библиотеке нет игр" << endl;
				break;
			}
			
			cout << "\nНа данный момент у вас " << gameList.size() << " игр и среди них: \n";
			for (watchGames = gameList.begin(); watchGames != gameList.end(); ++watchGames)
			{
				cout << " " << *watchGames << endl;
			}
			break;
```